### ISSUE TITLE
Change DATE_FORMAT to include milliseconds

### ISSUE THREAD

**Issue by fubu** [amiv-eth/amivapi#126]
[Currently](https://github.com/amiv-eth/amivapi/blob/f539f15d8310ff94b42ec1fe6fee30320a20cd74/amivapi/settings.py) the DATE_FORMAT is set to `%Y-%m-%dT%H:%M:%SZ`, which prevents JS clients to natively use their [`Date.toISOString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) method to convert JS datetimes into strings for sending them with JSON. This is because JS always includes milliseconds in the ISO representation, even if they are set to 0:

```js
d = new Date();
d.setMilliseconds(0);
d.toISOString();  # still includes milliseconds!
```

To simplify API clients (mainly JS-based ones at the moment) I propose to change the DATE_FORMAT to `%Y-%m-%dT%H:%M:%S.%fZ`

(The optimal solution would be to allow *both* formats, but unfortunately Eve does a simple `datetime.datetime.strptime` parsing with the configured DATE_FORMAT on incoming datetime strings, instead of using a real parser - which was [proposed, but rejected](https://github.com/nicolaiarocci/eve/issues/83).)

**Comment by hermannsblum** [comment]
I don't know enough about JS, therefore a simple question: How does the code look for the state now where we don't include nano seconds?

In python it's just the difference from `time.isoformat()`(which is wrong anyway) to `time.format(format_string)` which I find not a reason to adapt the API to a specific language, but I don't knwo about JS ;)

**Comment by NotSpecial** [comment]
If you take a look at the the subset of ISO 8601 (which defines time formats) which is approved by the w3c [here](https://www.w3.org/TR/NOTE-datetime) the "fullest" verison of the timestring they support is:

>  Complete date plus hours, minutes, seconds and a decimal fraction of a
second
      YYYY-MM-DDThh:mm:ss.sTZD (eg 1997-07-16T19:20:30.45+01:00)

It just happens that we decided to support the "less" complete version while JSs ```toISOstring``` uses the "full" format. Nothing wrong with either choice. But since at least now JS will be the dominant client to access the API and our choice is "random" anyway, why not chance it?

I support @fubus proposal of switching our time setting to: ```%Y-%m-%dT%H:%M:%S.%fZ```, because this means changing only [one line in settings.py](https://github.com/amiv-eth/amivapi/blob/master/amivapi/settings.py#L42).

As far as I can see we never set a time string and just use with datetime objects internally anyways.

**Comment by fubu** [comment]
Correct, if you ever manually interact with the API, the only difference to now is that you'd need to add `.0` into your hand-written datetime string. Also, most other languages usually also encode the milliseconds, or have simple means to do that.

On a side note: I'm also not a big fan of "customizing" the API to clients/client languages, but as said before Eve is unfortunately kind of limited in this case. When I write APIs I usually accept multiple variants of ISO8601 formatted dates, e.g. `2016-04-28T17:28:35.123+02:00` (which is not possible with *both* DATE_FORMATs discussed here, as they only work with UTC datetimes).

**Comment by hermannsblum** [comment]
Actually I would not say that did the decision about our current date-format randomly as from a human point of view milliseconds are just useless clutter 😉. I did a little bit of googleing concerning our other platforms Android and iOS and apparently the "standard" format varies between version numbers and platforms and moon phases and everything 😃

I am still curious how much code it actually safes a JS programmer, but of course we can change it.
